### PR TITLE
[REVIEW] dga detector refactor.

### PR DESCRIPTION
**Pull request by bsuryadevara**
- Added detector utils file to handle utility functions.
- Improved memory usage.
- Added detector utils test cases.
- Updated dga detector test cases.
- Updated dga detection notebook to reflect dga refactor.

### CODE REVIEW

**Comment by bsuryadevara** [comment]
Missed updating dga-detection notebook prior to pr creation. Updated just now.

**Comment by bsuryadevara** [comment]
Changed back to work in progress due to failures on clx nigthly container (root cause: torch version upgrade)

**Comment by bsuryadevara** [comment]
Note: pytorch 1.4.0 is assigning tasks to all accessible gpu's and incapable to deal with out of memory exception. This is happening because one of the gpu doesn't have enough memory. Whereas in pytorch 1.3.1 it would appear that when gpu memory is about full it is somehow preventing out of memory issue maybe by not assigning tasks.

**Comment by bsuryadevara** [comment]
dga detection process is crashing if available GPUs count is greater than 2 with pytorch version 1.4.0

File: clx/analytics/detector_utils.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import rmm
+import cudf
+import logging
+import numpy as np
+
+log = logging.getLogger(__name__)
+
+
+class DetectorDataset(object):
```
Comment (brhodes10) [inline clx/analytics/detector_utils.py]: Hi Bhargav, I'm concerned this class wrapper might be too thin of a wrapper holding just a list and count. Is there a way we can leverage this class a bit more? The utility functions seem useful, can we move some of that functionality to this class? Let me know if you want to discuss.

Comment (bsuryadevara) [reply clx/analytics/detector_utils.py]: Hi Bianca, Yea, I agree. We can have offline discussion.
I am thinking of creating private constructor for `DetectorDataset` and move `prepare_detector_dataset` function to DetectorDataset and make it as static. So, that dataset object can only be initiate with `prepare_detector_dataset `. 
do you think this is a good idea?

Comment (bsuryadevara) [reply clx/analytics/detector_utils.py]: @brhodes10 @efajardo-nv I have updated dga files as per the suggestion. thank you!

**Comment by brhodes10** [comment]
Thanks @bsuryadevara ! Can you also check to see if this file needs any updates https://github.com/rapidsai/clx/blob/branch-0.13/docs/source/intro-clx-dga.ipynb should be minor updates if any.

**Comment by bsuryadevara** [comment]
> Thanks @bsuryadevara ! Can you also check to see if this file needs any updates https://github.com/rapidsai/clx/blob/branch-0.13/docs/source/intro-clx-dga.ipynb should be minor updates if any.

notebook updated. thank you!

**Pull request merged**